crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -153,7 +153,28 @@ impl Analyzer<'_, '_> {
 
             {
                 match ty.normalize() {
-                    Type::Ref(_) => {
+                    Type::Ref(ref_ty) => {
+                        if let Some(ty_args) = &ref_ty.type_args {
+                            if !ty_args.params.is_empty() {
+                                if let RTsEntityName::Ident(id) = &ref_ty.type_name {
+                                    if let Ok(Some(ty)) = &self.find_type(&id.into()) {
+                                        let ty_found = &ty.clone().map(|v| v.into_owned()).collect::<Vec<Type>>()[0];
+
+                                        if !ty_found.span().is_dummy() && !ty_found.is_any() && ty_found.get_type_param_decl().is_none() {
```

Comment:
Span check seems wrong to me

Reply:
I guess my line of thinking was that you can't be trying to pass a generic to a type of dummy context but I guess that was wishful thinking.

My main motivation for adding this was that for
```ts
class D<T> {
  source: T;
  recurse: D<T>;
  wrapped: D<D<T>>;
}
```
I get

```rs
&ty_found = StaticThis(
    StaticThis {
        span: Span {
            lo: BytePos(
                0,
            ),
            hi: BytePos(
                0,
            ),
            ctxt: #0,
        },
        metadata: StaticThisMetadata {
            common: CommonTypeMetadata {
                implicit: false,
                infected_by_this_in_object_literal: false,
                prevent_converting_to_children: false,
                contains_infer_type: false,
                prevent_complex_simplification: false,
                resolved_from_var: false,
                prevent_generalization: false,
                destructure_key: DestructureId(
                    0,
                ),
            },
        },
        tracker: Tracker {
            _priv: (),
        },
    },
)
```

which I can't seem to be able to match against. for example all these are false:

```rs
dbg!(
    &ty_found.is_global_this(),
    &ty_found.is_this(),
    &ty_found.is_structured_or_instantiable(),
    &ty_found.is_builtin_interface("this"),
    &ty_found.is_builtin_interface("StaticThis"),
    &ty_found.is_builtin_interface("This"),
    &ty_found.is_builtin_interface("ThisType"),
    &ty_found.is_ref_type()
);
```

Reply:
nvm
```rs
if let Type::StaticThis(_) = ty_found.normalize() {
```
works.... :P